README.md

```diff
@@ -1,10 +1,260 @@
-# adminrouter
-The admin router is an open-source Nginx configuration created by Mesosphere that provides central authentication and proxy to DCOS services within the cluster.
+# Admin Router
 
+The Admin Router is an open-source Nginx configuration created by
+Mesosphere that provides central authentication and proxy to DC/OS services
+within the cluster.
 
 <img src="admin-router.png" alt="" width="100%" align="middle">
 
-## Ports
+## Ports summary
 <img src="admin-router-table.png" alt="" width="100%" align="middle">
- 
- 
+
+## Testing
+
+Admin Router repository includes a test harness that is meant to make
+testing easier and in some cases - possible. It's written in Python and
+uses pytest fixtures and custom modules to mock out all relevant DC/OS
+features and control Nginx startup and termination.
+
+All the tests are executed in a Docker container which is controlled by the
+Makefile. Inside the container pytest command is started which in turn pulls
```

Comment:
All tests are executed in a Docker container. This life cycle of this container is managed by the Makefile. Inside the container, pytest is executed and initiates the test run. On startup pytest pulls in all the relevant fixtures, such as a syslog mock, mocker (ie. a mock for DC/OS endpoints), dns mock, etc. Once the mocks are ready a Nginx instance is spawned using the developer's Admin Router configuration. This configuration is bind mounted from the developer's repository. Tests may launch several Nginx instances, in different configurations, depending on what is needed. Once the tests finish running pytest cleans up the environment and terminates any processes it spawned.